pkg/service/dbaas/service.go

```diff
@@ -120,17 +135,17 @@ func (s *Service) fetchManagedDBaaS(ctx *Context) error {
 				DBName: managedResource.GetName(),
 				Type:   groupVersionResource.Resource,
 			}
-			if organization, exist := managedResource.GetLabels()[service.OrganizationLabel]; exist {
-				dbaasDetail.Organization = organization
-			} else {
-				// cannot get organization from DBaaS
-				log.Info("Organization label is missing in DBaaS service, skipping...",
-					"label", service.OrganizationLabel,
-					"dbaas", managedResource.GetName())
-				continue
-			}
 			if namespace, exist := managedResource.GetLabels()[service.NamespaceLabel]; exist {
+				organization, ok := namespaces[namespace]
+				if !ok {
+					// cannot find namespace in namespace list
+					log.Info("Namespace not found in namespace list, skipping...",
+						"namespace", namespace,
+						"dbaas", managedResource.GetName())
+					continue
+				}
 				dbaasDetail.Namespace = namespace
+				dbaasDetail.Organization = organization
 			} else {
 				// cannot get namespace from DBaaS
 				log.Info("Namespace label is missing in DBaaS, skipping...",
```

Comment:
Very Nitpicky Nitpick :sweat_smile: : I think this loop is getting a bit hard to read with all the if .. else .. continue.  I'd probably be a bit easier to understand with a helper function and early returns. But it's not that bad yet and it's also fine as is